crates/compression/src/lib.rs

```diff
@@ -4,56 +4,87 @@
 #![deny(warnings)]
 
 pub mod compress;
+mod compressed_block_payload;
 pub mod config;
 pub mod decompress;
 mod eviction_policy;
 pub mod ports;
 mod registry;
 
 pub use config::Config;
+use enum_dispatch::enum_dispatch;
 pub use registry::RegistryKeyspace;
 
+use crate::compressed_block_payload::v0::CompressedBlockPayloadV0;
+#[cfg(feature = "fault-proving")]
+use crate::compressed_block_payload::v1::CompressedBlockPayloadV1;
```

Comment:
Why do we feature-gate this? The crate should probably just support all versions all the time.

Reply:
this would make my work easier, I can remove the feature gate np

Reply:
addressed in [9e19ab5](https://github.com/FuelLabs/fuel-core/pull/2551/commits/9e19ab582a0e851ff838e798ee3e583a75a1ee46)

Reply:
based on the discussion, we will keep the feature flag until all the sub-issues are sorted out